### PR TITLE
Remove unnecessary files and fix bug in particle to MPI rank assignment (ParticleSpatialLayout)

### PR DESCRIPTION
**Pull request by srikrrish**

### CODE REVIEW

**Comment by srikrrish** [comment]
This still needs to be tested!

**Comment by srikrrish** [comment]
This is ready now! The Tuple is fixed thanks to @manuel5975p and I tested again. Everything seems to compile and work fine.

File: README.md [hunk 1]
```diff
@@ -8,6 +8,9 @@ All IPPL releases (< 3.2.0) are available under the BSD 3-clause license. Since
 
 ## Installing IPPL and its dependencies
 
+All the new developments of IPPL are merged in to the `master` branch which can make it potentially unstable from time to time. So if you want a stable and more tested version
```
Comment (matt-frey) [inline README.md:11]: "in to" --> "into"

File: README.md [hunk 2]
```diff
@@ -8,6 +8,9 @@ All IPPL releases (< 3.2.0) are available under the BSD 3-clause license. Since
...
+please checkout the tagged branch correspodning to the last release (e.g. `git checkout tags/IPPL-x.x.x`). Otherwise if you want the latest developments go with the master with the above caveat in mind.
```
Comment (matt-frey) [inline README.md:12]: "last" --> "latest"

**Review by matt-frey — CHANGES_REQUESTED** [review]
Looks generally good. Two minor things.

**Review by matt-frey — APPROVED** [review]
Thanks for doing this!

**Review by aaadelmann — APPROVED** [review]

**Pull request merged**